presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,315 @@
+                statistics.getIntegerStatistics().ifPresent(integerStatistics -> {
+                    integerStatistics.getMin().ifPresent(catalogLongData::setMinimumValue);
+                    integerStatistics.getMax().ifPresent(catalogLongData::setMaximumValue);
+                });
+                statistics.getNullsCount().ifPresent(catalogLongData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogLongData::setNumberOfDistinctValues);
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.LONG.toString());
+                catalogColumnStatisticsData.setLongColumnStatisticsData(catalogLongData);
+                break;
+            case VARCHAR:
+            case CHAR:
+            case STRING:
+                StringColumnStatisticsData catalogStringData = new StringColumnStatisticsData();
+                statistics.getNullsCount().ifPresent(catalogStringData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogStringData::setNumberOfDistinctValues);
+                catalogStringData.setMaximumLength(statistics.getMaxValueSizeInBytes().orElse(0));
+                catalogStringData.setAverageLength(getAverageColumnLength(statistics.getTotalSizeInBytes(), rowCount, statistics.getNullsCount()).orElse(0));
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.STRING.toString());
+                catalogColumnStatisticsData.setStringColumnStatisticsData(catalogStringData);
+                break;
+            default:
+                throw new PrestoException(HIVE_INVALID_METADATA, "Invalid column statistics type: " + statistics);
+        }
+        return catalogColumnStatisticsData;
+    }
+
+    private static DecimalNumber bigDecimalToGlueDecimal(BigDecimal decimal)
+    {
+        Decimal hiveDecimal = new Decimal((short) decimal.scale(), ByteBuffer.wrap(decimal.unscaledValue().toByteArray()));
+        DecimalNumber catalogDecimal = new DecimalNumber();
+        catalogDecimal.setUnscaledValue(ByteBuffer.wrap(hiveDecimal.getUnscaled()));
+        catalogDecimal.setScale((int) hiveDecimal.getScale());
+        return catalogDecimal;
+    }
+
+    private static Optional<BigDecimal> glueDecimalToBigDecimal(DecimalNumber catalogDecimal)
+    {
+        Decimal decimal = new Decimal();
+        decimal.setUnscaled(catalogDecimal.getUnscaledValue());
+        decimal.setScale(catalogDecimal.getScale().shortValue());
+        return Optional.of(new BigDecimal(new BigInteger(decimal.getUnscaled()), decimal.getScale()));
+    }
+
+    private static Optional<LocalDate> dateToLocalDate(Date date)
+    {
+        long daysSinceEpoch = date.getTime() / DAY_TO_MILLISECOND_FACTOR;
+        return Optional.of(LocalDate.ofEpochDay(daysSinceEpoch));
```

Comment:
Someone more familiar with datetime gotchas should take a look at this.

Reply:
We may use  ```LocalDate.ofInstant``` but I'm not sure how to handle the ZoneId parameter